lib/routemaster/responses/response_promise.rb

```diff
@@ -1,16 +1,16 @@
-require 'concurrent/future'
+require 'concurrent/promise'
 require 'concurrent/executor/cached_thread_pool'
 require 'singleton'
 require 'delegate'
 
 module Routemaster
   module Responses
-    class FutureResponse
+    class ResponsePromise < Concurrent::Promise
```

Comment:
I'm not comfortable inheriting here. The original code was wrapping the future instead of inheriting; this allowed us to change libraries (from `thread` to `concurrent-ruby`) just recently, because we could guarantee that no on was depending on the interface provided by the other library (just on _our_ interface).

I'd like to do the same here: façade rather than inheritance, and expose jsut the bits we want, through delegation. Presumbably just `on_success` and `on_error`, and/or `then` but with a simpler interface.

In other words, I think we should avoid leaking the details of how we implement promises; both because it give us better maintainability, and removes an opportunity for shooting ourselves in the foot with concurrency issues.

Does that work for you?